pyzebra/ccl_findpeaks.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+import scipy as sc
+from scipy.interpolate import interp1d
+from scipy.signal import savgol_filter
+
+
+def ccl_findpeaks(
+    data, keys, int_threshold=None, prominence=None, smooth=True, window_size=None, poly_order=None
+):
+
+    """function iterates through the dictionary created by load_cclv2 and locates peaks for each measurement
+    args:   data (dictionary from load_cclv2),
+
+            int_threshold - fraction of threshold_intensity/max_intensity, must be positive num between 0 and 1
+                        i.e. will only detect peaks above 75% of max intensity
+
+            prominence - defines a drop of values that must be between two peaks, must be positive number
+                        i.e. if promimence is 20, it will detect two neigbouring peaks of 300 and 310 intesities, 
+                        if none of the itermediate values are lower that 290
+
+            smooth  - if true, smooths data by savitzky golay filter, if false - no smoothing
+
+            window_size - window size for savgol filter, must be odd positive integer
+
+            poly_order =  order of the polynomial used in savgol filter, must be positive integer smaller than 
+            window_size returns: dictionary with following structure:
+                                D{M34{  'num_of_peaks': 1,              #num of peaks
+                                        'peak_indexes': [20],           # index of peaks in omega array
+                                        'peak_heights': [90.],          # height of the peaks (if data vere smoothed 
+                                                                        its the heigh of the peaks in smoothed data)
+    """
+
+    if type(data) is not dict and data["file_type"] != "ccl":
+        print("Data is not a dictionary or was not made from ccl file")
+
+    int_threshold = 0.75 if int_threshold is None else int_threshold
+    prominence = 50 if prominence is None else prominence
+    smooth = False if smooth is None else smooth
+    window_size = 7 if window_size is None else window_size
+    poly_order = 3 if poly_order is None else poly_order
+
+    if 0 <= int_threshold <= 1:
+        pass
+    else:
+        int_threshold = 0.75
+        print(

```

Comment:
If you consider that the parameter is invalid, then `raise ValueError()`. The function should be aborted, rather than continued.

Reply:
Well I though it would be better this way - if the value is wrong, than default is selected instead, function can run and user knows that the input was wrong. I understand, that GUI will not have output, but can be easily changed to fit your needs. I was thinking that instead of print('value is wrong, set to xx'), you could put some messagebox with similar comment. Let me know what you think, I can change it.

Reply:
I see now, well, I don't have a strong opinion on changing bad parameters to their defaults, but I still have a few comments:
* Default value for `int_threshold` is 0.8 in the function and 0.75 in the check.
* It's clearer to check for a negative condition, rather than `pass` on a positive one and do the work inside `else`.

Reply:
Both corrected. I must admit that now, when I can see the original code and the new one, I'm starting to see a bit the "programming" thinking. Thank you!